crates/typst-library/src/layout/page.rs

```diff
@@ -148,6 +148,47 @@ pub struct PageElem {
     #[ghost]
     pub margin: Margin,
 
+    /// The page's bleed margin.
+    ///
+    /// The bleed is the area of content that extends beyond the final trimmed
+    /// size of the page. It ensures that no unprinted edges appear in the final
+    /// product, even if minor trimming misalignments occur.
+    ///
+    /// Accepted values:
+    ///
+    /// - `{auto}`: Sets the bleed to `0mm` on all sides.
```

Comment:
Since `auto` has no special meaning, it shouldn't be permitted. To get rid of it, you'll probably need to factor out part of the `Margin` struct. Maybe it's enough to remove the `_: AutoValue` branch and turn margin into `margin: Smart<Margin>`. I'm not 100% sure off the top of my head how this will interact with `#[fold]`, might need to double-check that it does not change the behavior.

Reply:
I think there was a misunderstanding here. I didn't want to change the existing behaviour of `margin` (allowing auto in the dictionary). I just wanted to prevent directly accepting `bleed: auto`. Perhaps that's a little tricky to achieve with good code reuse, but that would make the most sense conceptually imo.

Reply:
I'll go with this approach of using a generic for Margin so it only behaves differently in the case of bleed.